### PR TITLE
[fix]: [OIP-674]: Run DSL Tests inside testSuite to prevent port conf…

### PR DESCRIPTION
**Pull request by AbhijithHarness**
…lict

## Describe your changes
1. DSL tests are moved to a Test Suite, so that Hoverfly is started only once for all classes.
2. Names of Test classes are changed, so that they dont get picked up, outside of TestSuit
3. Renaming of files that uses class names

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51168)

### CODE REVIEW

**Comment by AbhijithHarness** [comment]
trigger uts

**Comment by AbhijithHarness** [comment]
trigger uts

**Comment by AbhijithHarness** [comment]
trigger messagecheck

**Comment by AbhijithHarness** [comment]
trigger messagecheck

**Review by naveen-harness — APPROVED** [review]

**Comment by AbhijithHarness** [comment]
trigger ut1

**Comment by AbhijithHarness** [comment]
trigger ut0

**Comment by AbhijithHarness** [comment]
trigger ut7

**Comment by AbhijithHarness** [comment]
trigger ut1

**Comment by AbhijithHarness** [comment]
trigger ut7

**Comment by n00bitax** [comment]
trigger ut1

**Comment by AbhijithHarness** [comment]
trigger ut1

**Comment by AbhijithHarness** [comment]
trigger ut1

File: srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/core/dsl/DSLHoverflyTestSuiteChildBase.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.reflect.MethodUtils;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+
+@Slf4j
+public abstract class DSLHoverflyTestSuiteChildBase extends CategoryTest {
```
Comment (dhruv-harness) [inline srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/core/dsl/DSLHoverflyTestSuiteChildBase.java:30]: nit: child and base seem contradictory, perhaps we can name it parent?

**Review by dhruv-harness — APPROVED** [review]

**Pull request merged**